p2p/libp2p/networksharding/listsSharder.go

```diff
@@ -243,7 +257,12 @@ func (ls *listsSharder) splitPeerIds(peers []peer.ID) map[int]sorting.PeerDistan
 		case core.ValidatorPeer:
 			peerDistances[intraShardValidators] = append(peerDistances[intraShardValidators], pd)
 		case core.ObserverPeer:
-			peerDistances[intraShardObservers] = append(peerDistances[intraShardObservers], pd)
+			shouldAppendToFullHistory := peerInfo.PeerSubType == core.FullHistoryObserver && ls.maxFullHistoryObservers > 0
```

Comment:
This condition is ok? If maxFullHistoryObservers would be at least 1, you would add here all the peers with PeerSubType = FullHistoryObserver to peerDistances[fullHistoryObservers] ?

Reply:
Yes, and then out of all the appended peers, the list sharder will choose only one, which is closer. All other peers will get disconnected from self.